src/messaging.ts

```diff
@@ -0,0 +1,144 @@
+import { fromEvent, Observable } from 'rxjs'
+import { filter, map, take } from 'rxjs/operators'
+import { UpdateExtensionSettingsArgs } from './context'
+
+/**
+ * One end of the client-page connection. The browser extension is a client, and
+ * the web app is a page.
+ */
+export type Source = 'Client' | 'Page'
+
+/** Useful for building tagged union types. */
+interface Tagged<T> {
+    type: T
+}
```

Comment:
What is this interface useful for if all the subinterfaces redeclare the `type` inline?

Reply:
Ah, I think this was more useful in a previous version of the code, but useless now. Removed.